### ISSUE TITLE
Reproducibility of BERT on SST-2 dataset

### ISSUE THREAD

**Issue by tingkai-zhang** [allenai/allennlp#2475]
Hi! I was wondering what is the accuracy you get using `Bert token Embedder`?
I add a classification layer on top of `Bert oken Embedder`, which is the same as the `BertForSentenceClassification` in huggingface repo.
I have tried both `bert-base-uncased` and `bert-base-cased`
The best test accuracy I get is 91.927%, the best validation accuracy is 93.224%
The original paper says the performance on SST-2 is 93.5%.

Also, I was wondering if anyone reproduce the result in original paper under Allennlp framework? 
Could you share the model and json file you used to reproduce the result?

**Comment by DeNeutoy** [comment]
That seems very close to the correct result.

Here are some ideas to get the extra 1%:
- Run a grid search
- Change the dropout you are using to `InputVariationalDropout`
- Use a `LearningRateSchedule` to decrease the learning rate towards the end of training
- Use an exponential weighted average of the final parameters
- Run some random seeds and see what the variance is
- Initialise your classification layer using either `xavier_init` or `orthogonal`


Hope that helps! Also note that the bert token embedder truncates sequences to some number of tokens (I can't remember what it is off the top of my head). Perhaps you have some long sequences which are being truncated?

**Comment by tingkai-zhang** [comment]
Thank you for your advice!
I tried some of the points you said before I opened this issue. There are two things I didn't try and I don't quite understand this one: 

>  Use an exponential weighted average of the final parameters

Could you give me a link on how that works?
I googled it but found something seems irrelevant.

**Comment by DeNeutoy** [comment]
see an example here: https://github.com/allenai/allennlp/blob/master/allennlp/training/moving_average.py

or alternatively you can just average the parameters written out on the `*.th` files manually 😄

**Comment by danishpruthi** [comment]
@tingkai-zhang I have a question, when you talk about SST-2 dataset and its performance, do you mean phrase level predictions (which have close to 65K phrases, and their binary labels), or the sentence level predictions which have close to 6.7K training instances?

**Comment by tingkai-zhang** [comment]
@danishpruthi  Hi, I am talking about sentence level binary classification task.
